llm_jailbreaking_defense/models.py

```diff
@@ -200,20 +200,30 @@ def __init__(
         self.batch_size = batch_size
         self.add_system_prompt = add_system_prompt
         self.template = template
+        self.quantization_config = quantization_config
+        self.system_message_template = system_message_template
 
         assert model_name is not None or preloaded_model is not None
         if preloaded_model is None:
             self.model, self.template = load_indiv_model(
-                model_name, max_memory=max_memory, load_in_8bit=load_in_8bit)
+                model_name, max_memory=max_memory, load_in_8bit=quantization_config.load_in_8bit)
         else:
             self.model = preloaded_model
             assert template is not None or model_name is not None
             if self.template is None:
                 _, self.template = get_model_path_and_template(model_name)
 
+
     def get_response(self, prompts_list, verbose=True, **kwargs):
         batch_size = len(prompts_list)
-        convs_list = [conv_template(self.template) for _ in range(batch_size)]
+        if self.system_message_template is None:
+            convs_list = [conv_template(self.template) for _ in range(batch_size)]
+        else:
+            convs_list = []
+            for _ in range(batch_size):
+                conv = get_conv_template(self.template)
```

Comment:
Shall we use `conv_template` instead of `get_conv_template` here?

Reply:
Yes you're right. I had issues with conv_template with an earlier implementation, but I just tested and I think conv_template will be fine.  I'll make those changes now.